nav2_behavior_tree/plugins/condition/would_a_controller_recovery_help_condition.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_behavior_tree/plugins/condition/would_a_controller_recovery_help_condition.hpp"
+#include <memory>
+
+namespace nav2_behavior_tree
+{
+
+WouldAControllerRecoveryHelp::WouldAControllerRecoveryHelp(
```

Comment:
Yeah, I like this better. Now this could be a funnel to whether to execute a sub-branch to do other contextual recoveries or the basic stuff we have now. Check if the error code is even something we'd need to check for specific codes for specific actions to help separate more cleanly the subtree tasks.